typescript/cli/src/context/strategies/signer/MultiProtocolSignerManager.ts

```diff
@@ -0,0 +1,121 @@
+import { Signer } from 'ethers';
+
+import {
+  ChainName,
+  ChainSubmissionStrategy,
+  MultiProvider,
+} from '@hyperlane-xyz/sdk';
+
+import { ENV } from '../../../utils/env.js';
+
+import { IMultiProtocolSigner } from './BaseMultiProtocolSigner.js';
+import { MultiProtocolSignerFactory } from './MultiProtocolSignerFactory.js';
+
+/**
+ * @title MultiProtocolSignerManager
+ * @dev Context manager for signers across multiple protocols
+ */
+export class MultiProtocolSignerManager {
+  private signerStrategies: Map<ChainName, IMultiProtocolSigner> = new Map();
+
+  constructor(
+    submissionStrategy: ChainSubmissionStrategy,
+    private chains: ChainName[],
+    private multiProvider: MultiProvider,
+    private key?: string,
+  ) {
+    // Initialize chain-specific strategies
+    for (const chain of chains) {
+      const strategy = MultiProtocolSignerFactory.getSignerStrategy(
+        chain,
+        submissionStrategy,
+        multiProvider,
+      );
+      this.signerStrategies.set(chain, strategy);
+    }
+  }
+
+  /**
+   * @dev Gets signers config for specified chains
+   */
+  private async getSignersConfig(): Promise<
+    Array<{ chain: ChainName; privateKey: string }>
+  > {
+    return Promise.all(
+      this.chains.map((chain) => this.getSignerConfigForChain(chain)),
+    );
+  }
+
+  /**
+   * @dev Gets private key from strategy or environment fallback
+   */
+  private async getSignerConfigForChain(
+    chain: ChainName,
+  ): Promise<{ chain: ChainName; privateKey: string }> {
+    const signerStrategy = this.signerStrategies.get(chain);
+    if (!signerStrategy) {
+      throw new Error(`No signer strategy found for chain ${chain}`);
+    }
+
+    // Determine private key with clear precedence
+    let privateKey: string;
+    if (this.key) {
+      privateKey = this.key;
```

Comment:
Maybe add some logging here so we know whether HYP_KEY or a strategy is being used